bokeh/core/_templates/autoload_nb_js.js

```diff
@@ -1,5 +1,116 @@
 {% extends "autoload_js.js" %}
 
+{% block register_mimetype %}
+
+  var MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
+  var CLASS_NAME = 'output_bokeh rendered_html';
+
+  /**
+   * Render data to the DOM node
+   */
+  function render(props, node) {
+    if (props.data['div'] !== undefined) {
+      var div = document.createElement("div");
+      div.innerHTML = props.data["div"];
+      node.appendChild(div);
+    }
+
+    if (props.data['script'] !== undefined) {
+      var script = document.createElement("script");
+      script.textContent = props.data["script"];
+      node.appendChild(script);
+    }
+  }
+
+  /**
+   * Handle when an output is cleared or removed
+   */
+  function handleClearOutput(event, handle) {
+    var cell = handle.cell;
+
+    var id = cell.output_area._bokeh_element_id;
+    var server_id = cell.output_area._server_id;
+    // Clean up Bokeh references
+    if (id !== undefined) {
+      Bokeh.index[id].remove();
+      delete Bokeh.index[id];
```

Comment:
Testing both, it still doesn't seem like the document is being appropriated removed. I'm unsure what the solution is here.